qemu/tests/thin_provisioning_image.py

```diff
@@ -0,0 +1,165 @@
+      :param env: Dictionary with test environment.
+      """
+
+    def count_block_num(image_disk):
+        """
+        Get the block num of the disk,when i create a large file in guest,the content needs
+        time to be written to the host disk,so when check using stat, you will find the block
+        number changes all the time for a period of time, this is related with the host env,
+        some machines is much faster,and some machines is slow, i test 3 host machines and finally
+        find when i set the wait_time to 12 seconds,it works fine..
+
+
+        :param image_disk: Image disk name.
+
+        :return: The block number after the status is steady
+        """
+        wait_time = 12
+        while True:
+            cmd = "stat {:s}".format(image_disk)
+            pattern = r"Blocks:\s(\d+)"
+            try:
+                block_num_1 = re.search(pattern, utils.system_output(cmd)).groups()[0]
+                time.sleep(wait_time)
+                block_num_2 = re.search(pattern, utils.system_output(cmd)).groups()[0]
+                logging.info("Disk is not steady yet, block_num: {:s}, block_num_after_10sec:{:s}".format(block_num_1,
+                                                                                                          block_num_2))
+                if block_num_1 == block_num_2:
+                    return block_num_1
+            except:
+                error.context("Got block number failed,Please Check.")
+
+    # Destroy all vms to avoid emulated disk marked drity before start test
+    thin_provisioning.destroy_vm(env)
+
+    vm_name = params["main_vm"]
+    params["start_vm"] = "yes"
+    enable_gluster = params.get("enable_gluster", "no")
+    base_dir = params.get("images_base_dir", data_dir.get_data_dir())
+    error.context("TEST STEP 1: Create test image disk.", logging.info)
+    tempdir = ""
+    gluster_volume = ""
+
+    if enable_gluster == "yes":
+        # TODO: product bug 1362730 exists , so can not run this case now.
+        gluster_volume = gluster.create_gluster_uri(params)
+        tempdir = auto_data_dir.get_tmp_dir()
+        gluster.glusterfs_mount(gluster_volume, tempdir)
+
+        params["only_disk_enable_gluster"] = "yes"
+        params["drive_format_gluster_disk"] = "scsi-hd"
+        params["drv_extra_params_gluster_disk"] = "discard=on"
+        params["image_name_gluster_disk"] = "gluster_disk"
+        params["force_create_image_gluster_disk"] = "no"
+        params["image_raw_device_gluster_disk"] = "yes"
+        image_file_name = "{}".format(gluster.get_image_filename(
+            params, params["image_name_gluster_disk"], params["image_format_gluster_disk"]))
+        params["image_name_gluster_disk"] = image_file_name
+        cmd = "qemu-img create -f {} {} {}".format(params["image_format_gluster_disk"],
+                                                   image_file_name, params["image_size_gluster_disk"])
```

Comment:
gluster has bug ,# TODO: product bug 1362730 exists , so can not run this case now.
and even if the bug is resolved ,since our gluster server is old ,if you use the default one ,it will fail. i debug using my own gluster server locally.